components/cli/pkg/runtime/runtime.go

```diff
@@ -54,33 +56,43 @@ type Runtime interface {
 	DeleteComponent(component SystemComponent) error
 	IsComponentEnabled(component SystemComponent) (bool, error)
 	CreateCelleryNameSpace() error
+	CreateNameSpace(namespace string) error
 	IsGcpRuntime() bool
 	CreatePersistentVolumeDirs() error
 	UpdateNfsServerDetails(ipAddress, fileShare string) error
 	UpdateMysqlCredentials(dbUserName, dbPassword, dbHost string) error
 	UpdateInitSql(dbUserName, dbPassword string) error
 	ApplyIstioCrds() error
 	InstallIstio() error
-	InstallIngressNginx(isLoadBalancerIngressMode bool) error
+	InstallIngressNginx(isLoadBalancerIngressMode bool, nodePortIpAddress string) error
 	ApplyKnativeCrds() error
 	InstallKnativeServing() error
 	InstallController() error
 	InstallMysql(isPersistentVolume bool) error
 	CreateConfigMaps() error
-	AddApim(isPersistentVolume bool) error
+	AddApim(isPersistentVolume bool, nfs Nfs, db MysqlDb) error
+	DeleteApim() error
 	AddObservability() error
-	AddIdp() error
+	AddIdp(db MysqlDb) error
+	DeleteIdp() error
 	UpdateNodePortIpAddress(nodePortIpAddress string) error
 	CreatePersistentVolume(hasNfs bool) error
 	IsHpaEnabled() (bool, error)
 	WaitFor(checkKnative, hpaEnabled bool) error
 	Validate() error
+	UnmarshalHelmValues(chartName string) error
```

Comment:
Shall we rename these 2 functions to something like "SetRuntimeValues", "GetRuntimeValues" to make it more descriptive? WDYT?

Reply:
+1 for the change.